packages/components/src/internal/components/editable/LookupCell.tsx

```diff
@@ -42,325 +74,105 @@ export interface LookupCellProps {
-            this.setState({
-                activeOptionIdx: index,
-            });
-        }
-    };
-
+export class LookupCell extends ReactN.Component<LookupCellProps, undefined, GlobalAppState> {
     isMultiValue = (): boolean => {
         return this.props.col.isJunctionLookup();
     };
 
-    onContainerScroll = (): void => {
-        // Issue 43051: LookupCell menu manually updated to account for scrolled grid container
-        if (this.menuEl.current && this.wrapperEl.current) {
-            const rect = this.wrapperEl.current.getBoundingClientRect();
-            this.menuEl.current.style.left = `${rect.left + window.scrollX}px`;
-            this.menuEl.current.style.top = `${rect.bottom}px`;
-        }
-    };
-
-    onInputBlur = (): void => {
-        this.blurTO = window.setTimeout(() => {
-            const { colIdx, modelId, rowIdx } = this.props;
-            this.props.select(modelId, colIdx, rowIdx);
-            this.resetLookup();
-        }, 200);
-    };
-
-    onInputChange = (): void => {
-        clearTimeout(this.changeTO);
-        this.changeTO = window.setTimeout(() => {
-            let token;
-
-            if (this.inputEl && this.inputEl.current) {
-                token = this.inputEl.current.value;
+    onInputChange = (fieldName: string, formValue: string | any[], items: any, selectedItems: Map<string, any>): void => {
+        const { colIdx, modelId, rowIdx, onCellModify } = this.props;
+        if (this.isMultiValue())
+        {
+            if (items.length == 0) {
+                modifyCell(
+                    modelId,
+                    colIdx,
+                    rowIdx,
+                    undefined,
+                    MODIFICATION_TYPES.REMOVE_ALL
+                );
             }
-
-            this.searchLookup(token);
-
-            this.setState({
-                activeOptionIdx: -1,
-                token,
-            });
-        }, 350);
-    };
-
-    onInputKeyDown = (event: KeyboardEvent<HTMLInputElement>): void => {
-        const { colIdx, modelId, rowIdx, values, onCellModify } = this.props;
-        const { activeOptionIdx } = this.state;
-        const options = this.getOptions(this.props);
-
-        switch (event.keyCode) {
-            case KEYS.Backspace:
-                if (!this.hasInputValue() && values !== undefined && values.size) {
-                    modifyCell(modelId, colIdx, rowIdx, values.last(), MODIFICATION_TYPES.REMOVE);
-                    if (onCellModify) onCellModify();
-                }
-                break;
-            case KEYS.Enter:
-                if (this.state.activeOptionIdx > -1) {
-                    this.onItemClick(options.get(activeOptionIdx));
-                } else {
-                    this.clearInput();
-                    this.props.select(modelId, colIdx, rowIdx + 1);
+            else {
+                const valueDescriptors = []
+                for (let i = 0; i < items.length; i++) {
```

Comment:
Could this be simplified some with a `.map()`? 

Something like this (untested):
```
const valueDescriptors = items.map(item => ({ raw: item.value, display: item.label })).toArray();
```